src/sql/workbench/services/queryEditor/browser/queryEditorService.ts

```diff
@@ -91,6 +91,8 @@ export class QueryEditorService implements IQueryEditorService {
 		const m = await fileInput.resolve();
 		if (sqlContent) {
 			m.textEditorModel.setValue(sqlContent);
+			// prevent backup and restore of invalid sql upon restart.
+			m.setDirty(false);
```

Comment:
This probably isn't going to solve the issue if the user edits the query (which while sort of an edge case is something we allow).

A quick and dirty solution would be to hook into the onDidChangeDirty event and set it to false.

A better solution would probably be to create a new input for edit data that always returns false for isDirty

Reply:
How do you hook onto the onDidChangeDirty event? Apparently you can only access the event itself, but not set it to set isDirty to false from queryEditorService

Reply:
I tried it and it worked fine for me - 
```
m.onDidChangeDirty( () => {
   m.setDirty(false);
}
```

(note this does create a bit of a feedback loop since changing the dirty state will cause the event to fire again. But that should only happen once since setting the dirty state to the same value doesn't fire the event again. May want to consider adding a guard to only do that if dirty state is true though regardless just to be safe in case the underlying implementation changes)

Reply:
It did not fire for me at all, it kept on reopening for me even when I tried to make an edit on a table and closing ADS